src/arch/arm/64/object/objecttype.c

```diff
@@ -163,7 +163,7 @@ finaliseCap_ret_t Arch_finaliseCap(cap_t cap, bool_t final)
     case cap_page_upper_directory_cap:
 #ifdef AARCH64_VSPACE_S2_START_L1
 #ifdef CONFIG_ARM_SMMU
-        if (cap_page_upper_directory_cap_get_capPGDMappedCB(cap) != CB_INVALID) {
+        if (cap_page_upper_directory_cap_get_capPUDMappedCB(cap) != CB_INVALID) {
```

Comment:
Does the top-level not depend on configuration? This looks like it would now break on other configs.

Reply:
Ah, I see the `AARCH64_VSPACE_S2_START_L1` guard now, sorry. The change should be correct as is then, please ignore.